src/main/java/de/unistuttgart/bugfinder/configuration/ConfigurationService.java

```diff
@@ -0,0 +1,96 @@
+
+@Service
+@Slf4j
+public class ConfigurationService {
+
+  @Autowired
+  private CodeService codeService;
+
+  @Autowired
+  private ConfigurationRepository configurationRepository;
+
+  @Autowired
+  private ConfigurationMapper configurationMapper;
+
+  @Autowired
+  private CodeMapper codeMapper;
+
+  public List<ConfigurationDTO> findAll() {
+    log.info("get all configurations");
+    return configurationMapper.toDTO(configurationRepository.findAll());
+  }
+
+  public Optional<ConfigurationDTO> find(final UUID id) {
+    log.info("get configuration {}", id);
+    return configurationMapper.toDTO(configurationRepository.findById(id));
+  }
+
+  public ConfigurationDTO save(final ConfigurationDTO configurationDTO) {
+    log.info("create configuration {}", configurationDTO);
+    return configurationMapper.toDTO(configurationRepository.save(configurationMapper.fromDTO(configurationDTO)));
+  }
+
+  public ConfigurationDTO delete(final UUID id) {
+    log.info("delete configuration {}", id);
+    Configuration configuration = configurationRepository
+      .findById(id)
+      .orElseThrow(() ->
+        new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Configuration with id %s not found.", id))
+      );
+    configurationRepository.deleteById(id);
+    return configurationMapper.toDTO(configuration);
+  }
+
+  public CodeDTO addCode(final UUID id, final CodeDTO code) {
+    log.info("add code {} to configuration {}", code, id);
+    final Configuration configuration = configurationRepository
+      .findById(id)
+      .orElseThrow(() ->
+        new ResponseStatusException(NOT_FOUND, String.format("Unable to find configuration with id %s", id))
+      );
+    final CodeDTO savedCode = codeService.save(code);
+    configuration.addCode(codeMapper.fromDTO(savedCode));
+    configurationRepository.save(configuration);
+    return savedCode;
+  }
+
+  public CodeDTO removeCode(final UUID id, final UUID codeId) {
+    log.info("remove code {} from configuration {}", codeId, id);
+    final Configuration configuration = configurationRepository
+      .findById(id)
+      .orElseThrow(() ->
+        new ResponseStatusException(NOT_FOUND, String.format("Unable to find configuration with id %s", id))
+      );
+    final CodeDTO persistedCode = codeService
+      .find(codeId)
+      .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, String.format("Unable to find code with id %s", id)));
+    configuration.removeCode(codeMapper.fromDTO(persistedCode));
+    configurationRepository.save(configuration);
+    return null;
+  }
+
+  public List<CodeDTO> getCodes(final UUID id) {
+    log.info("get codes from configuration {}", id);
+    final Configuration configuration = configurationRepository
+      .findById(id)
+      .orElseThrow(() ->
+        new ResponseStatusException(NOT_FOUND, String.format("Unable to find configuration with id %s", id))
+      );
+    return codeMapper.toDTO(configuration.getCodes());
```

Comment:
```suggestion
    log.debug("get all configurations");
    return configurationMapper.toDTO(configurationRepository.findAll());
  }

  public Optional<ConfigurationDTO> find(final UUID id) {
    log.debug("get configuration {}", id);
    return configurationMapper.toDTO(configurationRepository.findById(id));
  }

  public ConfigurationDTO save(final ConfigurationDTO configurationDTO) {
    log.debug("create configuration {}", configurationDTO);
    return configurationMapper.toDTO(configurationRepository.save(configurationMapper.fromDTO(configurationDTO)));
  }

  /**
   * Deletes the configuration with the given ID, if present.
   *
   * @param id the ID of the configuration to delete
   * @return the deleted configuration, if found
   * @throws ResponseStatusException (204 - NO_CONTENT) if the configuration was not found
   */
  public ConfigurationDTO delete(final UUID id) {
    log.info("delete configuration {}", id);
    Configuration configuration = configurationRepository
      .findById(id)
      .orElseThrow(() -> new ResponseStatusException(HttpStatus.NO_CONTENT)));
    configurationRepository.deleteById(id);
    return configurationMapper.toDTO(configuration);
  }

  public CodeDTO addCode(final UUID id, final CodeDTO code) {
    log.debug("add code {} to configuration {}", code, id);
    final Configuration configuration = getConfiguration(id);
    final CodeDTO savedCode = codeService.save(code);
    configuration.addCode(codeMapper.fromDTO(savedCode));
    configurationRepository.save(configuration);
    return savedCode;
  }

  public CodeDTO removeCode(final UUID id, final UUID codeId) {
    log.debug("remove code {} from configuration {}", codeId, id);
    final Configuration configuration = getConfiguration(id);
    final CodeDTO persistedCode = codeService
      .find(codeId)
      .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, String.format("Unable to find code with id %s", id)));
    configuration.removeCode(codeMapper.fromDTO(persistedCode));
    configurationRepository.save(configuration);
    return persistedCode;
  }

  public List<CodeDTO> getCodes(final UUID id) {
    log.debug("get codes from configuration {}", id);
    return codeMapper.toDTO(getConfiguration(id).getCodes());
  }
    
  /**
   * @return the configuration with the given ID
   * @throws ResponseStatusException (404) if no configuration with this ID exists
   */
  private Configuration getConfiguration(final UUID id) {
    return configurationRepository
      .findById(id)
      .orElseThrow(() ->
        new ResponseStatusException(NOT_FOUND, String.format("Unable to find configuration with id %s", id))
      );
    }
```
Regarding the change in the delete method: No need to throw a `404` if all you want to do is deleting it…